x-pack/plugins/spaces/public/copy_saved_objects_to_space/lib/process_import_response.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type {
+  SavedObjectsImportAmbiguousConflictError,
+  SavedObjectsImportConflictError,
+  SavedObjectsImportFailure,
+  SavedObjectsImportMissingReferencesError,
+  SavedObjectsImportResponse,
+  SavedObjectsImportSuccess,
+  SavedObjectsImportUnknownError,
+  SavedObjectsImportUnsupportedTypeError,
+} from 'src/core/public';
+
+export interface FailedImport {
+  obj: Omit<SavedObjectsImportFailure, 'error'>;
+  error:
+    | SavedObjectsImportConflictError
+    | SavedObjectsImportAmbiguousConflictError
+    | SavedObjectsImportUnsupportedTypeError
+    | SavedObjectsImportMissingReferencesError
+    | SavedObjectsImportUnknownError;
+}
+
+export interface ProcessedImportResponse {
+  success: boolean;
+  failedImports: FailedImport[];
+  successfulImports: SavedObjectsImportSuccess[];
+}
+
+// This is derived from the function of the same name in the savedObjectsManagement plugin
+export function processImportResponse(
+  response: SavedObjectsImportResponse
+): ProcessedImportResponse {
+  const { success, errors = [], successResults = [] } = response;
+  const failedImports = errors.map<FailedImport>(({ error, ...obj }) => ({ obj, error }));
+  return {
+    success,
+    failedImports,
+    successfulImports: successResults,
+  };
+}
```

Comment:
I needed to remove this to handle a cyclic dependency between the `spaces` and `savedObjectsManagement` plugin.

Since copy-to-space was implemented as a thin veneer over export/import, the CTS flyout used this existing `processImportResponse` function from the `savedObjectsManagement` plugin to process the import response.

Turns out we could pare this down quite a bit, we didn't need a lot of what the original function provided. So I decided to just recreate a CTS-specific function here to remove that problematic dependency :+1:

Reply:
Seems like the most pragmatic approach